### PR TITLE
191 add people to friends

### PR DESCRIPTION
**Pull request by potapchuk-anna**

### CODE REVIEW

File: frontend/src/app/modules/social-page/social-page/social-page.component.ts [hunk 1]
```diff
@@ -10,14 +11,22 @@ import { UserFilter } from '../../../models/filters/userFilter';
...
+    public constructor(
+        private userService: UserService,
+        private friendHub: FriendHubService,
+    ) {
     }
 
-    ngOnInit(): void {
+    async ngOnInit(): Promise<void> {
         this.users$ = this.userService.getUsers();
+        await this.friendHub.start();
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/social-page/social-page/social-page.component.ts:25]: Please... Change it to http requests

File: frontend/src/app/modules/user-card/user-card.component.html [hunk 2]
```diff
@@ -44,15 +44,61 @@
...
+<!--                <span *ngIf="isRequest">Accept</span>-->
+<!--            </button>-->
+<!--            <button class="button" [disabled]="isFollowDisabled()" (click)="friendClick()">-->
+<!--                <img-->
+<!--                    [src]="!isRequest ? 'assets/user-card-icons/follow-icon.svg' : 'assets/user-card-icons/cancel.svg'"-->
+<!--                    alt="book" />-->
+<!--                <span *ngIf="!isRequest">Follow</span>-->
+<!--                <span *ngIf="isRequest">Cancel</span>-->
+<!--            </button>-->
+<!--        </div>-->
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/user-card/user-card.component.html:102]: Remove comment

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs [hunk 3]
```diff
@@ -158,4 +185,33 @@ public async Task<UserDto> MakeAdminAsync(int userId)
...
+        var friendshipsWithUsers = _context.Friends.Include(f => f.User).Include(f => f.Requester);
+        var requestors = await friendshipsWithUsers
+            .Where(f => f.FriendshipStatus != FriendshipStatus.Rejected && f.UserId == _authService.UserId)
+            .Select(f=>f.Requester)
+            .ToListAsync();
+
+        var acceptors = await friendshipsWithUsers
+            .Where(f => f.FriendshipStatus != FriendshipStatus.Rejected && f.RequesterId == _authService.UserId)
+            .Select(f => f.User)
+            .ToListAsync();
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs:207]: What about something like
```suggestion
        var users= await friendshipsWithUsers
            .Where(f => f.FriendshipStatus != FriendshipStatus.Rejected && f.UserId == _authService.UserId ||  f.RequesterId == _authService.UserId)
            .Select(f=>f.UserId == _authService.UserId ? f.Requester : f.User)
            .ToListAsync();

```

Comment (Anatolii-Ivanov) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs:207]: You can also add status here (as tuple, for example) UserFriendshipStatus

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs [hunk 4]
```diff
@@ -90,14 +88,43 @@ public async Task<List<UserShortInfoDto>> GetFilteredUsers(UserFilterDto userFil
...
+                user.UserFriendshipStatus = UserFriendshipStatus.Acceptor;
+            }
+            else if (confirmedFriends.Any(friend => friend.UserId == user.Id || friend.RequesterId == user.Id))
+            {
+                user.UserFriendshipStatus = UserFriendshipStatus.Friend;
+            }
+            else
+            {
+                user.UserFriendshipStatus = UserFriendshipStatus.Regular;
+            }
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs:119]: I'm not really sure if it makes sense to show users on social page...
And 3 SQL requests is a bit too much for setting status...

**Review by Anatolii-Ivanov — COMMENTED** [review]
Good job, but I don't understand why you decided to make real time update here...

File: frontend/src/app/modules/user-card/user-card.component.ts [hunk 5]
```diff
@@ -29,4 +43,23 @@ export class UserCardComponent implements OnInit {
...
+        this.friendService.acceptFriendship({ email: this.user.email! })
+            .subscribe(() => {
+                this.user.userFriendshipStatus = UserFriendshipStatus.Friend;
+            });
+    }
+
+    onRejectClicked() {
+        this.friendService.acceptFriendship({ email: this.user.email! }).subscribe(() => {
+            this.user.userFriendshipStatus = UserFriendshipStatus.Regular;
+        });
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/user-card/user-card.component.ts:63]: Shouldn't you send different models? Or at least different endpoint

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs [hunk 6]
```diff
@@ -72,7 +71,11 @@ public async Task<List<UserShortInfoDto>> GetFilteredUsers(UserFilterDto userFil
     {
         var users = _context.Users
             .Include(u => u.Tags)
-            .Include(u => u.Image);
+            .Include(u => u.Image)
+            .Where(u => !_context.Friends.Any(f=>
+            (f.UserId == _authService.UserId || f.RequesterId == _authService.UserId)
+            && (f.UserId == u.Id || f.RequesterId == u.Id))
+            && u.Id != _authService.UserId);
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs:78]: Check tabulation

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs [hunk 7]
```diff
@@ -158,4 +188,26 @@ public async Task<UserDto> MakeAdminAsync(int userId)
...
+    {
+        var friendshipsWithUsers = _context.Friends.Include(f => f.User).Include(f => f.Requester);
+        var users = await friendshipsWithUsers
+           .Where(f => f.FriendshipStatus != FriendshipStatus.Rejected && f.UserId == _authService.UserId || f.RequesterId == _authService.UserId)
+           .Select(f => f.UserId == _authService.UserId ? f.Requester : f.User)
+           .ToListAsync();
+
+        var mappedFriends = _mapper.Map<List<UserShortInfoDto>>(users);
+
+        await FillUserFriendshipStatus(mappedFriends);
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs:209]: Last question
Can't you fill it with executed query?

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by bochka123 — APPROVED** [review]

**Pull request merged**